app/static/src/app/components/options/NumericInput.vue

```diff
@@ -69,10 +75,80 @@ export default defineComponent({
+                // wrap tooltip props update with update to input box
+                // as the v-tooltip directive needs an update to actually
+                // hide the tooltip
+                textValue.value = `${textValue.value} `;
+                tooltipProps.value.content = "";
+                textValue.value = textValue.value.trim();
             }
         };
 
+        const validateArray = (arr: BoundTooltip[], numeric: number, ascending: boolean) => {
+            arr.sort((x, y) => {
+                if (x.number < y.number) return ascending ? -1 : 1;
+                if (x.number > y.number) return ascending ? 1 : -1;
+                return 0;
+            });
+
+            const condition = (compareNum: number) => {
+                return ascending ? numeric > compareNum : numeric < compareNum;
+            };
+
+            arr.forEach((bound) => {
+                if (condition(bound.number)) {
+                    tooltipProps.value.content = bound.message;
+                    tooltipProps.value.variant = bound.variant || "error";
+                }
+            });
+        };
+
+        const minMaxValidation = (numeric: number) => {
+            if (typeof props.maxAllowed === "number") {
+                if (numeric > props.maxAllowed) {
+                    tooltipProps.value.content = `Please enter a value no greater than ${props.maxAllowed}`;
+                    return props.maxAllowed;
+                }
+                tooltipProps.value.content = "";
+            }
+
+            if (typeof props.minAllowed === "number") {
+                if (numeric < props.minAllowed) {
+                    tooltipProps.value.content = (props.minAllowed === 0)
+                        ? "Please enter a non-negative number"
+                        : `Please enter a value no less than ${props.minAllowed}`;
+                    return props.minAllowed;
+                }
+                tooltipProps.value.content = "";
+            }
+
+            if (Array.isArray(props.maxAllowed) && props.maxAllowed.length > 0) {
+                const maxesAllowed = [...props.maxAllowed] as BoundTooltip[];
+                validateArray(maxesAllowed, numeric, true);
+
+                if (numeric > maxesAllowed.at(-1)!.number) {
+                    return maxesAllowed.at(-1)!.number;
+                }
+                if (numeric <= maxesAllowed[0].number) {
+                    tooltipProps.value.content = "";
+                }
+            }
+
+            if (Array.isArray(props.minAllowed) && props.minAllowed.length > 0) {
+                const minsAllowed = [...props.minAllowed] as BoundTooltip[];
+                validateArray(minsAllowed, numeric, false);
+
+                if (numeric < minsAllowed.at(-1)!.number) {
+                    return minsAllowed.at(-1)!.number;
+                }
+                if (numeric >= minsAllowed[0].number) {
+                    tooltipProps.value.content = "";
```

Comment:
This logic (without elses) does mean that the container could do something stupid like:
`minAllowed: [10], maxAllowed: [5]` and value '7' would show no tooltip, but value would be set to 10.
Given those values are coming from config, I wonder if we should actually show an error or at least log an error if props are nonsensical.

Same would go for numeric type min/max.

Reply:
hmm this is a general problem for all min maxes yes! i am not sure what the best way to do this would be? we could log and error in console or what i am leaning towards is we display a tooltip that is always on an says min prop is greater than max prop or something like that? its like a user facing console for those that may not check the console?

Reply:
My instinct is that we probably shouldn't surface the issue to the user, but yeah like you say the app creator is more likely to quickly run up the app and spot an always on tooltip than check the console. 

Tbh, I feel like that might be too late and we should maybe have a config validation stage, though not clear when that would be. 

A problem for another ticket!